lib/redis_test.rb

```diff
@@ -74,10 +74,11 @@ def start
     end
 
     def stop
-      %x{
-        cat #{pidfile} | xargs kill -QUIT
-        rm -f #{cache_path}#{db_filename}
-      }
+      if File.file?(pidfile) && File.readable?(pidfile)
+        pid = File.read(pidfile).to_i
+        Process.kill("QUIT", pid) if pid > 0
+      end
```

Comment:
Another alternative:
```
pid = File.read(pidfile).to_i rescue 0
Process.kill("QUIT", pid) if pid > 0
```
Shorter, but bare-rescue could be considered bad manners.